### ISSUE TITLE
[HOLD] [$500] [MEDIUM] Split bill – When Smartscan fails in Workspace, the Merchant field is empty and the amount is displayed as $0.00.

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36046]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.38-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**  Applause internal team
**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Login
3. Navigate to the workspace chat
4. Click on the + button > Split bill
5. Use a receipt that will not fail the scan
6. Continue to the final review page
7. Create the expense
8. When the scan request is complete click on the split preview to navigate to the split report 

## Expected Result:
The amount and merchant fields have been updated

## Actual Result:
When Smartscan fails in Workspace, the Merchant field is empty and the amount is displayed as $0.00.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/f19b6c2b-eef8-493d-9a84-c3bc49423c90

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f4dd13f508eb66a6</li>
        <li>Upwork Job ID: 1755275350909222912</li>
        <li>Last Price Increase: 2024-02-27</li>
    </ul>
</details>

**Comment by m-natarajan** [comment]
We think that this bug might be related #wave6-collect-submitters
CC @greg-schroeder

**Comment by abdulrahuman5196** [comment]
No proposals yet

**Comment by abekkala** [comment]
https://expensify.slack.com/archives/C03UK30EA1Z/p1707851230196889

**Comment by abekkala** [comment]
https://expensify.slack.com/archives/C04878MDF34/p1708013831428379

**Comment by abekkala** [comment]
https://app.slack.com/client/E047TPA624F/C04878MDF34

**Comment by brunovjk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Split bill – Merchant field is empty, amount is $0.00 when Smartscan fails in WS:
1. Splitbill receipt does not carry value and merchant correctly.
2. When Splitbill receipt does not load value and merchant correctly, the same fields continue with 0.00 and empty merchant

### What is the root cause of that problem?
1. I believe it is a backend issue, I tried several different receipts, and none of them managed to extract the value correctly.
2. In `startSplitBill` and `completeSplitBill` we hardcode the value of amount and merchant to `0` and `"none"`:
https://github.com/Expensify/App/blob/4ae7b293018a7cd3d41a265cdca34f1dd8417552/src/libs/actions/IOU.ts#L2000 https://github.com/Expensify/App/blob/4ae7b293018a7cd3d41a265cdca34f1dd8417552/src/libs/actions/IOU.ts#L2007
https://github.com/Expensify/App/blob/4ae7b293018a7cd3d41a265cdca34f1dd8417552/src/libs/actions/IOU.ts#L2325  And if the value is not loaded correctly at the time of scanning, 0.00 and "" remains.

### What changes do you think we should make in order to solve the problem?
1. Check at BE why we haven't received the correct amount on the receipt.
2. Update `completeSplitBill` to show another message/value if the scan is unsuccessful, instead of leaving 0 and "".

### What alternative solutions did you explore? (Optional)

**Comment by kmbcook** [comment]
I don't think this issue is described correctly. In the video, a receipt is used which FAILS, which is the opposite of what step 5 describes. Which is correct? And, the actual result description is missing.

**Comment by trjExpensify** [comment]
@m-natarajan - please can you include: 

1. the account email from the above
2. the receiptID (open the expense in OldDot and type this into the JS console: `g_megaEdit.expense.receiptID`)
3. the file used for upload

Here’s going through splitting a receipt with a workspace that I know won’t fail: 

https://github.com/Expensify/App/assets/16232057/3e96acfe-08e9-4f58-ae60-820019bba908

**Comment by abekkala** [comment]
Holding as I think this may not be an issue. 
The failed smartscan outcome above is expected. 
Once I get the info above, I can look into the receipt ID, etc

**Comment by abekkala** [comment]
I don't see a report under that ID: 8639230505798598

If the smartscan failed and provided a red "Merchant" and $0.00 - **that is expected and not a bug**

**Comment by rojiphil** [comment]
@abekkala I think the issue here is that there is no violation message displayed for `amount` even though amount is 0 and the smart scan has failed to process the amount. However, violation message is displayed for missing `merchant`.

And I think the root cause for this is that we are not handling `receipt` state of `DELETED` sent by BE (refer below). To resolve this in FE, we can add `DELETED` to `RECEIPT_STATE` [here](https://github.com/Expensify/App/blob/539e269ef024f691591904f85ab4072bf4d4c166/src/CONST.ts#L1282C1-L1288C11) and include this state in the condition [here](https://github.com/Expensify/App/blob/4a885f596547a16893ae40ebabcff41c82416453/src/pages/iou/SplitBillDetailsPage.js#L94). Alternatively, if it was not intentional to send `DELETED` state, BE can also send `SCANFAILED` state which will resolve the problem here. But if we are doing this in FE, I can put this in proposal format. 

I just had a look at this issue. But it looks like the party's already over here. Maybe not!
Do you think this can be reopened?
cc @trjExpensify @abdulrahuman5196 

![Screenshot 2024-02-22 at 9 46 02 PM](https://github.com/Expensify/App/assets/3004659/2de20e18-7e9f-453b-abd4-64fd845dc526)

**Comment by kbecciv** [comment]
I'm reopening the issue for a second review. Previously, there was an error next to the Amount, Merchant, and Split button. 
![Screenshot 2024-02-27 at 22 05 33 (2)](https://github.com/Expensify/App/assets/93399543/ada7cee1-c7e6-4965-bf78-7d04e67bc259)

**Comment by rojiphil** [comment]
Converting my comment [here](https://github.com/Expensify/App/issues/36046#issuecomment-1961081301) into proposal format. 
What are your thoughts on this?
cc @abdulrahuman5196 

## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The issue here is that there is no violation message displayed for `amount` even though amount is 0 and the smart scan has failed to process the amount. However, violation message is displayed for missing `merchant`.

### What is the root cause of that problem?

The root cause for this is that we are not handling `receipt` state of `DELETED` sent by BE (refer below).

![Screenshot 2024-02-22 at 9 46 02 PM](https://github.com/Expensify/App/assets/3004659/2de20e18-7e9f-453b-abd4-64fd845dc526)

### What changes do you think we should make in order to solve the problem?

To resolve this in FE, we can add `DELETED` to `RECEIPT_STATE` [here](https://github.com/Expensify/App/blob/539e269ef024f691591904f85ab4072bf4d4c166/src/CONST.ts#L1282C1-L1288C11) and include this state in the condition [here](https://github.com/Expensify/App/blob/4a885f596547a16893ae40ebabcff41c82416453/src/pages/iou/SplitBillDetailsPage.js#L94). Alternatively, if it was not intentional to send `DELETED` state, BE can also send `SCANFAILED` state which will resolve the problem here. 

### What alternative solutions did you explore? (Optional)

**Comment by trjExpensify** [comment]
Sorry @kbecciv but for us to actually triage this issue and look at evaluate proposals accurately the OP needs to be improved a lot. For example: 

- Look at the "actual result" listed, it provides zero insight into what actually happened.

>**Actual result:**
>Describe what actually happened

- Step 5 in the action steps state to use a receipt that **will not fail the scan**

>5. On the scan option, upload a digital receipt that will not fail the scan

Contrast that to the title which alludes to "when SmartScan fails in WS". 

- Step 8 in the action steps alludes to navigating to a "report conversation" but if that was supposed to mean the split report, a split report has no conversation.

>When the scan request is complete click on the split preview to navigate to the report conversation

Overall, this is a pretty poor issue. Can we start over and clean this up to then revaluate accurately, please? Thanks!

**Comment by kbecciv** [comment]
@trjExpensify Apologies for the oversight. I've made the necessary corrections in the OP based on your suggestion.

**Comment by abekkala** [comment]
@kbecciv 
I still need some clarification here: 

You state :

> 5. Use a receipt that will not fail the scan


yet, in your video I'm not seeing that SmartScan has been given the time to actually complete and be considered a successful smartscanned receipt. 
<img width="336" alt="Screenshot 2024-02-27 at 3 24 26 PM" src="https://github.com/Expensify/App/assets/11966460/a01a3be4-61e2-4ac4-942a-dce499fb5cb2">

Often, SmartScan does NOT complete in a matter of 2-3 seconds. 

Have you tested this with a SmartScanned receipt that actually completes a successful smartscan . 

From what I can tell you're attempting to split a receipt that does not have a successful SmartScan and that outcome is expected.

**Comment by abekkala** [comment]
And going back to your title: **When Smartscan fails in Workspace, the Merchant field is empty and the amount is displayed as $0.00**

This is expected and how SmartScan operates

**Comment by abekkala** [comment]
I've changed this back to $500 as I prematurely increased as I misunderstood the original issue as originally provided. 

IF this moves forward (although I'm still not convinced this is a real issue) this can go back to the original price after discussions have been complete to pinpoint IF there is a real issue and that issue should be stated correctly and in detail.

**Comment by brunovjk** [comment]
@abekkala Can you please tell me what should happen "When Smartscan fails in Workspace"? The Merchant field should be empty and the amount should be $0.00?

**Comment by brunovjk** [comment]
Great! Thank you. Just one more thing please: If a SmartScan fails, should both failure message for the merchant and amount be rendered?

<details>
<summary>I tested in the last main now:</summary>



![image](https://github.com/Expensify/App/assets/95647348/d477ecf4-a256-44c4-a150-dba5836ac377)

</details>

**Comment by abekkala** [comment]
yes, here is an example from a failed smartscan just yesterday: 
<img width="670" alt="Screenshot 2024-02-27 at 3 55 50 PM" src="https://github.com/Expensify/App/assets/11966460/b62b403e-4d8e-445c-9562-82442f1944d0">

**Comment by brunovjk** [comment]
Great! Thank you. What about the amount failure message?
![image](https://github.com/Expensify/App/assets/95647348/bfd690cf-640f-4b36-8763-699ac60d7f8c)

**Comment by abekkala** [comment]
A successful SmartScan requires at least an amount & date. Without either one of those two things, the SmartScan will fail every time.
For context, if there is not a merchant, or it cannot be clearly read, “Unknown Merchant” will be entered (non parsed receipt).

@trjExpensify in the case of requesting money I believe we expect something different, correct? Meaning smartscanning an expense receipt (for reporting purposes) means something different than than when smartscanning for an IOU. Based on[ this convo](https://expensify.slack.com/archives/C05RECHFBEW/p1707248843965969?thread_ts=1707244879.273649&cid=C05RECHFBEW) that you and Jason participated in.

**Comment by brunovjk** [comment]
Are those related https://github.com/Expensify/App/issues/34916#issuecomment-1905293484? Thank yoiu.

**Comment by trjExpensify** [comment]
>@trjExpensify in the case of requesting money I believe we expect something different, correct? Meaning smartscanning an expense receipt (for reporting purposes) means something different than than when smartscanning for an IOU. Based on[ this convo](https://expensify.slack.com/archives/C05RECHFBEW/p1707248843965969?thread_ts=1707244879.273649&cid=C05RECHFBEW) that you and Jason participated in.

That hasn't been implemented yet for requests that go on iouReports (i.e "P2P, requests to friends"), so that wouldn't factor in here. It's being progressed by @youssef-lr [here](https://github.com/Expensify/Expensify/issues/345849) (internal repo). But yes, at some point in the future iouReport'd requests won't show a merchant field unless you scan a receipt successfully. 

>For context, if there is not a merchant, or it cannot be clearly read, “Unknown Merchant” will be entered (non parsed receipt).

Yeah, I think it's still expected in NewDot that:

1. "Unknown merchant" to be populated when the merchant field is deemed not clear on the receipt but `amount` and/or `date` are, such that it's a partial receipt. In this case, you would not see the `Enter a merchant name` error message. (CC: @Gonals)
2. The merchant field is blank when a receipt upload is marked as illegible (i.e say a random picture with no `merchant`, `amount` or `date` readable). 

>What about the amount failure message?

1. The `amount` field should be blank and has the error `Enter an amount` when the receipt is marked illegible.
2. The `amount` field has `$0.00` and has the `Enter an amount` error message when the receipt is deemed partial. 

CC: @mountiny as well for a fact check on these, as I know you've worked on this a bit at some point in the last few months. It's pretty easy to get turned around!

**Comment by JmillsExpensify** [comment]
Agreed, this is tricky. That sounds right to me though.

**Comment by rojiphil** [comment]
Based on the following expected behavior mentioned [here](https://github.com/Expensify/App/issues/36046#issuecomment-1968076880)

> 	1	The amount field has $0.00 and has the Enter an amount error message when the receipt is deemed partial.

and, the following comment [here](https://github.com/Expensify/App/issues/36046#issuecomment-1966643499) from the testing team (Also please note the attached image there) 

> I'm reopening the issue for a second review. Previously, there was an error next to the Amount, Merchant, and Split button.

it seems to me that one issue here is that the violation message is not displayed for `amount`. 
Looks like there is consensus coming out on this. right?

**Comment by rojiphil** [comment]
>	1	"Unknown merchant" to be populated when the merchant field is deemed not clear on the receipt but amount and/or date are, such that it's a partial receipt. In this case, you would not see the Enter a merchant name error message. (CC: [@Gonals](https://github.com/Gonals))
	2	The merchant field is blank when a receipt upload is marked as illegible (i.e say a random picture with no merchant, amount or date readable).

And this looks a little confusing to me. Especially since there is no reference to usage of `Unknown Merchant` in code in FE.
Or maybe the BE should have populated the merchant text with `Unknown Merchant` in which case FE would have naturally not shown the error message.
Looking forward to authentic feedback from @mountiny @gonals.

**Comment by brunovjk** [comment]
Would anyone have an example of a receipt that doesn't fail that they could send me? Mine are always failing. Thank you in advance :D

**Comment by abekkala** [comment]
When I tested in staging if I prematurely clicked `Split` prior to letting the SmartScan complete (it's still in progress)
an Amount error appears: 
<img width="683" alt="Screenshot 2024-02-29 at 5 30 42 PM" src="https://github.com/Expensify/App/assets/11966460/c2fd1e72-a398-4b8d-87ec-d837e063e348">

I let this SmartScan complete (it failed - not sure why as the example receipt does show a date, merchant, amount 🤷‍♀️)
In any case, I am provided the red dot 'error' once I expand the receipt there is no error and it only appears once I click `Split`
![2024-02-29_17-35-50 (1)](https://github.com/Expensify/App/assets/11966460/47eec580-326b-44a6-b070-60b5b67c6612)

**Comment by rojiphil** [comment]
> I let this SmartScan complete (it failed - not sure why as the example receipt does show a date, merchant, amount 🤷‍♀️)
> In any case, I am provided the red dot 'error' once I expand the receipt there is no error and it only appears once I click `Split`

@abekkala Exactly. And, that I think is the issue here.  Earlier it was showing error message but now it does not. 
On clicking on `Split` button though, the validation will anyway fail and show the error as expected.

**Comment by abekkala** [comment]
CC: @mountiny or @Gonals would you be able to fact check the above?

**Comment by mountiny** [comment]
Sorry, lots of comments in this issue. @rojiphil or someone else would you mind summarizing what are the questions here please?

**Comment by youssef-lr** [comment]
I can help investigating this one on Monday @mountiny

**Comment by rojiphil** [comment]
> Sorry, lots of comments in this issue.  [@rojiphil](https://github.com/rojiphil) or someone else would you mind summarizing what are the questions here please?

@mountiny Agree. Quite a lot of discussions happened here.
Not sure of others, but to me, it looks like there is no ambiguity about the `amount` issue. 
However, regarding `Unknown merchant`, the following comment [here](https://github.com/Expensify/App/issues/36046#issuecomment-1968076880) seems like inconsistent with the current FE.

> 1 "Unknown merchant" to be populated when the merchant field is deemed not clear on the receipt but amount and/or date are, such that it's a partial receipt. In this case, you would not see the Enter a merchant name error message. (CC: [@Gonals](https://github.com/Gonals))

Currently, there is no reference to the usage of Unknown Merchant in code in FE. So, it makes me wonder if the BE should have populated the merchant text with Unknown Merchant in which case FE would have naturally not shown the error message. Or is this a recent change that needs to be implemented in FE? 
cc @youssef-lr

**Comment by youssef-lr** [comment]
Going to run some tests and get back to you @rojiphil.

**Comment by abekkala** [comment]
@youssef-lr is there any update for @rojiphil on this one?

**Comment by youssef-lr** [comment]
I think the summary is still [this](https://github.com/Expensify/App/issues/36046#issuecomment-1968076880) and it sounds good to me as well. I think we just need to implement it? This will need backend changes I think because the frontend has no way of knowing if the receipt is marked as illegible.

**Comment by abekkala** [comment]
not overdue - still on hold

**Comment by youssef-lr** [comment]
Same as above.

**Comment by abekkala** [comment]
@twisterdotcom I'm ooo until April 08 - I'll be taking back any Open issues when I return

___

### CURRENT STATUS: 
On HOLD. If it comes off Hold before April 08 it will most likely require backend changes (Internal) see comment from @youssef-lr [here](https://github.com/Expensify/App/issues/36046#issuecomment-2000346047)

**Comment by twisterdotcom** [comment]
Still on HOLD apparently

**Comment by twisterdotcom** [comment]
@abekkala is back today.

**Comment by youssef-lr** [comment]
Still held.

**Comment by abekkala** [comment]
still on hold

**Comment by youssef-lr** [comment]
Still held.

**Comment by abekkala** [comment]
not overdue, on hold

**Comment by abekkala** [comment]
on hold

**Comment by youssef-lr** [comment]
Correct @abekkala, I'll try to get to this next week

**Comment by abekkala** [comment]
Still on hold

**Comment by abekkala** [comment]
@youssef-lr any updated info?

**Comment by youssef-lr** [comment]
I'm not able to prioritize this @abekkala since we paused working on vip-split, I think it can wait, I'm currently focusing on a issues from wave-collect, wave-control, and fast-apis.

**Comment by abekkala** [comment]
On HOLD

**Comment by abekkala** [comment]
still on hold for Youssef

**Comment by abekkala** [comment]
still on hold

**Comment by abekkala** [comment]
On hold

**Comment by youssef-lr** [comment]
Still held

**Comment by youssef-lr** [comment]
Still held.

**Comment by abekkala** [comment]
on hold

**Comment by abekkala** [comment]
Still on hold

**Comment by abekkala** [comment]
I'm going ooo until Oct 20 - I don't think this will be off hold before then, so I'll stay assigned as the only BZ member. 

@youssef-lr if it does go off-hold before then; 
1. please unassign me
2. remove Bug label
3 reapply Bug label (that will assign someone else)
4. then also re-assign me 🤪 

That will allow someone to help manage the progress until I return! 😄

**Comment by youssef-lr** [comment]
Still held.

**Comment by youssef-lr** [comment]
On hold.

**Comment by abekkala** [comment]
not overdue - still holding this one

**Comment by youssef-lr** [comment]
Same.

**Comment by abekkala** [comment]
@youssef-lr it's been a while so I want to check in again - you'd like to still Hold on this one?

**Comment by youssef-lr** [comment]
Correct.

**Comment by abekkala** [comment]
Still on HOLD Mel

**Comment by youssef-lr** [comment]
Same.

**Comment by youssef-lr** [comment]
Still held.

**Comment by youssef-lr** [comment]
Same.

**Comment by youssef-lr** [comment]
Same.
